src/modules/75-cf/components/FlagActivation/ServicesList.tsx

```diff
@@ -262,7 +257,7 @@ const ServicesList: React.FC<ServicesListProps> = props => {
               setServices(initialServices)
               setShowModal(false)
             }}
-            allServices={serviceData?.data?.serviceDeploymentDetailsList}
+            allServices={serviceData as ServiceResponseDTO[]}
```

Comment:
I dont think this is very safe cast - serviceData could be `undefined` and you're forcing it into a `ServiceResponseDTO` type

Which means if it is undefined, and you try to access its properties within the `EditServicesModal`, it'll break - even though typescript won't complain

I see you're defaulting the prop within `EditServicesModal` to an empty array, so you dont need to cast it here

```suggestion
            allServices={serviceData}
```

Reply:
without the cast this error comes up 
![image](https://user-images.githubusercontent.com/95680496/184853540-8f384909-2f4b-460a-aa28-0d766bdac322.png)

the `undefined` is due to this map 
```const serviceData = servicesResponse?.data?.content?.map(serviceContent => serviceContent.service)

Reply:
Ah understood.

if you replace your map with this, does it work remove the error?
```
  const serviceData = servicesResponse?.data?.content
    ?.filter(serviceContent => serviceContent.service !== undefined)
    .map(serviceContent => serviceContent.service as ServiceResponseDTO)
```

What this does it filter out the undefined services, then maps over the resulting array and picks out the services. The cast is safe in this case because you've already filtered out any undefined services in the previous map

Reply:
it does! thank you. Update that now